saw-central/src/SAWCentral/Crucible/Common/Vacuity.hs

```diff
@@ -0,0 +1,108 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module SAWCentral.Crucible.Common.Vacuity (
+  GenericCrucibleContext(..),
+  checkAssumptionsForContradictions
+) where
+
+import Prelude hiding (fail)
+import           Control.Lens
+import           Control.Monad (forM_)
+import           Control.Monad.Extra (findM, whenM)
+import           Data.Function
+import           Data.List
+import qualified What4.ProgramLoc as W4
+import qualified What4.Interface as W4
+import qualified Lang.Crucible.Backend as Crucible
+import SAWCore.SharedTerm
+import SAWCoreWhat4.ReturnTrip
+import SAWCentral.Proof
+import SAWCentral.TopLevel
+import SAWCentral.Value
+import SAWCentral.Options
+import qualified SAWCentral.Crucible.Common as Common
+import           SAWCentral.Crucible.Common.MethodSpec (CrucibleMethodSpecIR)
+import qualified SAWCentral.Crucible.Common.MethodSpec as MS
+
+type AssumptionReason = (MS.ConditionMetadata, String)
+
+-- | Minimal backend-independent context
+data GenericCrucibleContext sym = GenericCrucibleContext
+  { gccSym :: sym  -- What4 symbolic backend
+  }
```

Comment:
I don't understand what the purpose of this data type is. Why not just pass `Sym` directly instead?